### PR TITLE
alpaca actions #3549

### PR DESCRIPTION
**Pull request by feyzullah**
Listed triggers in the issue has been added as actions because mentioned triggers are to list info.
## Actions
* Cancel All Order
* Cancel Order
* Close All Postions
* Close Position
* Get Account Info
* Place Order
* List Orders
* List Positions

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4239"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/alpaca/actions/list-orders/list-orders.mjs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      params: {
+        symbols: this.symbols && this.symbols.join(),
+        side: this.side,
+        status: this.status,
+        limit: 500, //max allowed limit, no pagination
+        after: after && new Date(after).toISOString(),
+        until: until && new Date(until).toISOString(),
+        nested: this.nested,
+      },
+    });
```
Comment (michelle0927) [inline components/alpaca/actions/list-orders/list-orders.mjs:85]: I think you could use the `limit` and `after` properties to paginate through the results. You'd use the the `submitted_at` time of the page's last order as the `after` for the next request.

File: components/alpaca/alpaca.app.mjs [hunk 2]
```diff
@@ -1,11 +1,145 @@
...
+          params: {
+            limit: 500, //max allowed limit, no pagination
+          },
+        });
+        return orders.map((order) => ({
+          label: `${order.symbol} ${order.type} ${order.side} - ${order.qty ?? order.notional}`,
+          value: order.id,
+        }));
+      },
+    },
```
Comment (michelle0927) [inline components/alpaca/alpaca.app.mjs:44]: See previous comment. I think you could use the `limit` and `after` properties here as well to paginate through the orders.

**Review by michelle0927 — DISMISSED** [review]

File: components/alpaca/alpaca.app.mjs [hunk 3]
```diff
@@ -1,11 +1,158 @@
...
+        const after = prevContext.nextAfter;
+        const orders = await this.getOrders({
+          isPaperAPI,
+          params: {
+            limit: pageSize,
+            after,
+            direction: "asc",
+          },
+        });
+        const nextAfter = orders[orders.length - 1]?.updated_at;
```
Comment (andrewjschuang) [inline components/alpaca/alpaca.app.mjs:46]: In the last page when `nextAfter` becomes `undefined`, do the options start again from the beginning?

File: components/alpaca/actions/list-orders/list-orders.mjs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          status: this.status,
+          after: nextAfter,
+          until: until && new Date(until).toISOString(),
+          nested: this.nested,
+          direction: "asc",
+          limit: pageSize,
+        },
+      });
+      for (const order of response)
+        orders.push(order);
```
Comment (andrewjschuang) [inline components/alpaca/actions/list-orders/list-orders.mjs:89]: Does this work?

```suggestion
      orders.push(...response);
```

File: components/alpaca/actions/list-orders/list-orders.mjs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    nested: {
+      type: "string",
+      label: "Nested",
+      description: "If true, the result will roll up multi-leg orders under the `legs` field of primary order.",
+      optional: true,
+    },
+  },
+  async run ({ $ }) {
+    const after = this.after && Date.parse(this.after);
+    if (this.after && isNaN(after))
```
Comment (andrewjschuang) [inline components/alpaca/actions/list-orders/list-orders.mjs:67]: Although not mandatory, I would say adding brackets `{}` make it a lot more readable

**Review by andrewjschuang — COMMENTED** [review]
Hi @feyzullah, just some minor comments.
Moving to QA nonetheless.

File: components/alpaca/alpaca.app.mjs (hunk 3 above)
Comment (feyzullah) [reply components/alpaca/alpaca.app.mjs:46]: @andrewjschuang you are right I guess I forgot to add the `if` statement there

File: components/alpaca/actions/list-orders/list-orders.mjs (hunk 4 above)
Comment (feyzullah) [reply components/alpaca/actions/list-orders/list-orders.mjs:89]: it will definitely work, thank you

**Review by andrewjschuang — APPROVED** [review]

**Pull request merged**